src/vec32.rs

```diff
@@ -93,25 +93,38 @@ impl<T> Vec32<T> {
     /// Append an element to the vector.
     ///
     /// Panics if the number of elements in the vector overflows `u32`.
+    #[inline]
     pub fn push(&mut self, value: T) {
         if self.len == self.cap {
             self.reserve(1);
         }
         unsafe {
-            let end = self.as_mut_ptr().offset(self.len as isize);
+            let end = self.as_mut_ptr().add(self.len as usize);
             ptr::write(end, value);
             self.len += 1;
         }
     }
+    #[inline]
+    pub fn extend_from_slice(&mut self, other: &[T]) {
+        if self.len as usize + other.len() as usize >= self.cap as usize {
+            self.reserve(other.len() as u32);
```

Comment:
To handle the case where `other.len() > u32::MAX as usize` here, I'm wondering if we should do something more like this?

```rust
self.reserve(other.len().try_into().expect("capacity overflow"));
```

This would keep us consistent with the panic behavior of `Vec32::reserve` itself:

https://github.com/mbrubeck/mediumvec/blob/398db615123c5172f9e44aee42ca1dc19b7d6d2c/src/vec32.rs#L184